src/service/action-impl.js

```diff
@@ -468,8 +467,17 @@ export class ActionService {
       const invokeAction = () => {
         // Use `this.root_` as the target for global targets e.g. "AMP".
         // Otherwise, targetType should be an element id.
-        const target = (this.globalTargets_[targetType]) ?
+        let target = (this.globalTargets_[targetType]) ?
           this.root_ : this.root_.getElementById(targetType);
+        if (!target) {
+          const doc = this.root_;
+          var allNodes = doc.getElementsByTagName('*');
+          for (var i = 0; i < allNodes.length; i++) {
+            if(allNodes[i].shadowRoot) {
+              target = allNodes[i].shadowRoot.getElementById(targetType);
```

Comment:
Ditto.

Reply:
Currently: 
To see what the target for this action is we only check the elements in the document itself. However if the element being targetted is in the shadow document (as is the case with say a position observer in a page inserted via amp-next-page) we never find the target. 

This fixes that by seeing if there are potential targets in the shadow DOM

Reply:
same as above, both `position-observer` and `actions` services are doc-level services and in the `amp-next-page` example, `amp-animation`, amp-position-observer` and the actions are all happening in the same document. This should also never pierce shadow boundaries. The root cause is likely something else (wring root being passed somewhere)